open_spiel/algorithms/infostate_tree.cc

```diff
@@ -491,12 +491,57 @@ std::shared_ptr<InfostateTree> MakeInfostateTree(
     const Game& game, Player acting_player,
     int max_move_limit) {
   return std::shared_ptr<InfostateTree>(new InfostateTree(
-      game, acting_player, max_move_limit));
+      {game.NewInitialState().get()}, /*chance_reach_probs=*/{1.},
+      game.MakeObserver(kInfoStateObsType, {}),
+      acting_player, max_move_limit));
+}
+
+std::shared_ptr<InfostateTree> MakeInfostateTree(
+    const std::vector<InfostateNode*>& start_nodes,
+    int max_move_ahead_limit) {
+  std::vector<const InfostateNode*> const_nodes(
+      start_nodes.begin(), start_nodes.end());
+  return MakeInfostateTree(const_nodes, max_move_ahead_limit);
+}
+
+std::shared_ptr<InfostateTree> MakeInfostateTree(
+    const std::vector<const InfostateNode*>& start_nodes,
+    int max_move_ahead_limit) {
+  SPIEL_CHECK_FALSE(start_nodes.empty());
+  const InfostateNode* some_node = start_nodes[0];
+  const InfostateTree& originating_tree = some_node->tree();
+  SPIEL_DCHECK_TRUE([&](){
+    for (const InfostateNode* node : start_nodes) {
+      if (!node) return false;
+      if (!node->is_leaf_node()) return false;
+      if (node->depth() != some_node->depth()) return false;
+      if (&node->tree() != &originating_tree) return false;
+    }
+    return true;
+  }());
+
+  std::vector<const State*> start_states;
+  start_states.reserve(start_nodes.size() * 8);
+  std::vector<double> chance_reach_probs;
+  chance_reach_probs.reserve(start_nodes.size() * 8);
+
+  for (const InfostateNode* node : start_nodes) {
+    for (int i = 0; i < node->corresponding_states_size(); ++i) {
+      start_states.push_back(node->corresponding_states()[i].get());
+      chance_reach_probs.push_back(node->corresponding_chance_reach_probs()[i]);
+    }
+  }
+
+  return std::shared_ptr<InfostateTree>(new InfostateTree(
+      start_states, chance_reach_probs,
+      originating_tree.infostate_observer_,
+      originating_tree.acting_player_,
+      max_move_ahead_limit));
 }
 
 std::shared_ptr<InfostateTree> MakeInfostateTree(
     const std::vector<const State*>& start_states,
-    const std::vector<float>& chance_reach_probs,
+    const std::vector<double>& chance_reach_probs,
```

Comment:
Are you sure you need double precision? A `float` has 4-5 decimals' precision, `double` usually makes sense only when error can accumulate, like summing some values, subtracting similar values etc.

Reply:
I think it's better because of LP construction.